src/sidebar/service-context.tsx

```diff
@@ -71,16 +71,14 @@ export function withServices<
     // Inject services, unless they have been overridden by props passed from
     // the parent component.
 
-    const services: Record<string, unknown> = {};
+    const services: Partial<Record<ServiceName, unknown>> = {};
     for (const service of serviceNames) {
       // Debugging check to make sure the store is used correctly.
-      if (process.env.NODE_ENV !== 'production') {
-        if (service === 'store') {
-          /* istanbul ignore next - Ignore debug code */
-          throw new Error(
-            'Do not use `withServices` to inject the `store` service. Use the `useStore` hook instead',
-          );
-        }
+      if (process.env.NODE_ENV !== 'production' && service === 'store') {
+        /* istanbul ignore next - Ignore debug code */
+        throw new Error(
+          'Do not use `withServices` to inject the `store` service. Use the `useStore` hook instead',
+        );
```

Comment:
I just combined both nested `if`s into a single one

Reply:
Can you check that this if body still gets eliminated from the bundle in production builds. That might have been the reason for having separate blocks originally?

Reply:
Oh! I didn't know that it would get removed. Let me check.

Reply:
Yes, it does get removed from the production bundle.